h1st/model/fuzzy/fuzzy_variables.py

```diff
@@ -0,0 +1,123 @@
+        :param membership_funcs: this is the list of tuple(membership_func_name, membership_func_type, membership_func_range)
+            there are four different kinds of membership_func_type that are supported and their membership_func_range should follow the following formats.
+            GAUSSIAN: [mean, sigma]
+            SIGMOID: [offset, width]
+            TRIANGLE: [a, b, c] where a <= b <= c
+            TRAPEZOID: [a, b, c, d] where a <= b <= c <= d
+
+        .. code-block:: python
+            :caption: example
+            vars = FuzzyVariables()
+            vars.add(
+                var_name='var1',
+                var_type='antecedent',
+                var_range=np.arange(0, 15+1e-5, 0.5),
+                membership_funcs=[('normal', fm.GAUSSIAN, [3, 3.3]),
+                                  ('abnormal', fm.TRIANGLE, [8, 15, 15])]
+            )
+        """
+        # Check variable type.
+        if var_type == "antecedent":
+            fuzzy_var = skctrl.Antecedent(var_range, var_name)
+        elif var_type == "consequent":
+            fuzzy_var = skctrl.Consequent(var_range, var_name)
+        else:
+            logger.error(f"{var_type} is not supported type")
+            raise ValueError(f"{var_type} is not supported type")
+
+        # Add membership function and its values.
+        for mem_func_name, mem_func_type, mem_func_vals in membership_funcs:
+            if mem_func_type == fm.TRIANGLE:
+                if len(mem_func_vals) != 3:
+                    raise ValueError(
+                        (
+                            f"TRIANGLE membership function needs 3 "
+                            f"values. Provided {len(mem_func_vals)} values."
+                        )
+                    )
+                fuzzy_var[mem_func_name] = skfuzzy.trimf(
+                    fuzzy_var.universe, mem_func_vals
+                )
+            elif mem_func_type == fm.TRAPEZOID:
+                if len(mem_func_vals) != 4:
+                    raise ValueError(
+                        (
+                            f"TRAPEZOID membership function needs 4 "
+                            f"values. Provided {len(mem_func_vals)} values."
+                        )
+                    )
+                fuzzy_var[mem_func_name] = skfuzzy.trapmf(
+                    fuzzy_var.universe, mem_func_vals
+                )
+            elif mem_func_type == fm.GAUSSIAN:
+                if len(mem_func_vals) != 2:
+                    raise ValueError(
+                        (
+                            f"GAUSSIAN membership function needs 2 "
+                            f"values. Provided {len(mem_func_vals)} values."
+                        )
+                    )
+                fuzzy_var[mem_func_name] = skfuzzy.gaussmf(
+                    fuzzy_var.universe, mem_func_vals[0], mem_func_vals[1]
+                )
+            elif mem_func_type == fm.SIGMOID:
+                if len(mem_func_vals) != 2:
```

Comment:
This whole thing is too long and looks more complicated than it needs to be. Change it to doing a table look-up of the  required number of parameters corresponding to each MemberShip enum value.

Reply:
done